workerapi/api.go

```diff
@@ -98,9 +98,19 @@ func ThriftRunStatusToDomain(thrift *worker.RunStatus) runner.RunStatus {
 	if thrift.ExitCode != nil {
 		domain.ExitCode = int(*thrift.ExitCode)
 	}
+	if thrift.SnapshotId != nil {
+		domain.SnapshotID = *thrift.SnapshotId
+	}
 	return domain
 }
 
+func copyString(s string) *string {
```

Comment:
Do you mind double checking for this PR that the other thrift types are being converted correctly?